src/serializationHelpers.ts

```diff
@@ -402,32 +356,80 @@ export function serializeParameters(
     const typeTag: ParameterType = paramSchema?.typeTag ?? ParameterType.Unit;
     switch (typeTag) {
         case ParameterType.U8:
-            return encodeWord8(userInput as number);
+            if (typeof userInput === 'number') {
+                return encodeWord8(userInput as number);
+            } else {
+                throw new Error('Unsigned integer required');
+            }
         case ParameterType.U16:
-            return encodeWord16(userInput as number);
+            if (typeof userInput === 'number') {
+                return encodeWord16(userInput as number);
+            } else {
+                throw new Error('Unsigned integer required');
+            }
         case ParameterType.U32:
-            return encodeWord32(userInput as number);
+            if (typeof userInput === 'number') {
+                return encodeWord32(userInput as number);
+            } else {
+                throw new Error('Unsigned integer required');
+            }
         case ParameterType.U64:
-            return encodeWord64(BigInt(userInput));
+            if (typeof userInput === 'number') {
```

Comment:
@shjortConcordium We have tested update contract without  parameters working here is the txHash:e6032082e8244b2f24cd423b0bbe3350e2f6124a63b950a83256225d85c4004c, 
the input is Buffer.from([]) since we except no parameters
and update contract with enum type here is the txHash:0057b1313cd10cf9ebad686f5e5bc98ed8303a18a3fab311f09ea581753f1208
the input is const userInput = DCBBankState[DCBBankState.Intact];
how you are sending the input can we know?

Reply:
Okay, it was an issue on my end.
I was trying to update a contract called `PiggyBank`, and used receiveName:
`insert`, instead of `PiggyBank.insert`.
Though it is weird that that causes an incorrect serialization on chain and not just during parsing?

Reply:
@shjortConcordium  we found the issue and we will look into it. Thanks

Reply:
@shjortConcordium  we have fixed the enum init and update. Please validate and let us know.

Reply:
@shjortConcordium I tried to update two-step-transfer example it is failing with "Update contract failed: Rejected by contract logic with reason -5" what does this mean?(http://dashboard.testnet.concordium.com/lookup/1e8ff88445d5276451983a61f9c4425787f3a8dee5df84ecc6be9192c2f96e31).
I have sent input as const userInput = {
        RequestTransfer: [0, 5, '3gLPtBSqSi7i7TEzDPpcpgD8zHiSbWEmn23QZH29A7hj4sMoL5']
    };
For initiating the contract below is the input
const userInput = {
        account_holders: [
            '4ZJBYQbVp3zVZyjCXfZAAYBVkJMyVj8UKUNj9ox5YqTCBdBq2M',
            '3gLPtBSqSi7i7TEzDPpcpgD8zHiSbWEmn23QZH29A7hj4sMoL5'
        ],
        transfer_agreement_threshold: 2,
        transfer_request_ttl: 1643888516,
    };

Reply:
the -5 is the 5th reason in error Enum in:
https://github.com/Concordium/concordium-rust-smart-contracts/blob/main/examples/two-step-transfer/src/lib.rs#L145
Which is insufficient funds (either you haven't deposited enough, or its because of incorrect endian)

Reply:
@shjortConcordium  thanks for the update. Now I have send some CCD/GTU amount it is working amount: new GtuAmount(10000n),http://dashboard.testnet.concordium.com/lookup/b66d0a90f2dbb25c20eda9d7067229339968e34b2622726507386f4ddf2a56b6